extensions/provisioning/verify.go

```diff
@@ -428,27 +429,28 @@ func VerifySnapshots(client *rancher.Client, localclusterID string, clusterName
 			return false, fmt.Errorf("no snapshots found")
 		}
 
-		// Indexed from 0 for S3 checks to ensure that the local backup location does not have the s3Prefix.
-		// Needed to ensure that the correct S3 snapshot is restored.
-		if strings.Contains(snapshotNameList[0], s3Prefix) {
-			snapshotToBeRestored = snapshotNameList[len(snapshotNameList)-1]
-			return true, nil
-		}
+		if strings.Contains(fmt.Sprintf("%v", snapshotNameList), s3Prefix) {
+			snapshotSteveObjList, err := client.Steve.SteveType(stevetypes.EtcdSnapshot).List(nil)
+			if err != nil {
+				return false, err
+			}
 
-		if len(snapshotNameList) == expectedSnapshotLength {
-			snapshotToBeRestored = snapshotNameList[0]
-			return true, nil
+			for _, snapshot := range snapshotSteveObjList.Data {
+				if snapshot.Annotations["etcdsnapshot.rke.io/storage"] == "s3" {
```

Comment:
just confirming this annotation is there for rke2 and rke1 cluster types

Reply:
also is this worth throwing in a const?

Reply:
Correct. I thought of throwing in a const, but  I saw in this same file that we aren't consistently doing that for annotations. So I didn't initially. I can if we want, not a big deal.

Reply:
Resolved.